fast_com.py

```diff
@@ -1,222 +1,194 @@
+        if line.find('token:') >= 0:
+            if verbose:
+                print("line is", line)
+            token = line.split('"')[1]
+            if verbose:
+                print("token is", token)
+            if token:
+                break
+
+    baseurl = 'https://api.fast.com/'
+    if forceipv4:
+        # force IPv4 by connecting to an IPv4 address of
+        # api.fast.com (over ... HTTP)
+        ipv4 = findipv4('api.fast.com')
+        # HTTPS does not work IPv4 addresses, thus use HTTP
+        baseurl = 'http://' + ipv4 + '/'
+    elif forceipv6:
+        # force IPv6
+        ipv6 = findipv6('api.fast.com')
+        baseurl = 'http://[' + ipv6 + ']/'
+
+    # Not more than 3 possible
+    url = baseurl + 'netflix/speedtest?https=true&token=' + token
+    url += '&urlCount=3'
+    if verbose:
+        print("API url is", url)
+    try:
+        urlresult = requests.post(url, data=None, timeout=2)
+    except Exception as e:
+        print(e)
+        # not good
+        if verbose:
+            # probably IPv6, or just no network
+            print("No connection possible")
+            return 0    # no connection, thus no speed
+
+    jsonresult = urlresult.text
+    parsedjson = json.loads(jsonresult)
+
+    # Prepare for getting those URLs in a threaded way:
+    amount = len(parsedjson)
+    if verbose:
+        print("Number of URLs:", amount)
+    threads = [None] * amount
+    results = [0] * amount
+    urls = [None] * amount
+    i = 0
+    for jsonelement in parsedjson:
+        # fill out speed test url from the json format
+        urls[i] = jsonelement['url']
+        if verbose:
+            print(jsonelement['url'])
+        i = i + 1
+
+    # Let's check whether it's IPv6:
+    for url in urls:
+        fqdn = url.split('/')[2]
+        try:
+            socket.getaddrinfo(fqdn, None, socket.AF_INET6)
+            if verbose:
+                print("IPv6")
+        except:
+            pass
+
+    # Now start the threads
+    for i in range(len(threads)):
+        threads[i] = Thread(target=gethtmlresult, args=(urls[i], results, i))
+        threads[i].daemon = True
+        threads[i].start()
+
+    # Monitor the amount of bytes (and speed) of the threads
+    time.sleep(1)
+    sleepseconds = 3    # 3 seconds sleep
+    lasttotal = 0
+    highestspeedkBps = 0
+    nrloops = maxtime // sleepseconds
+    for loop in range(nrloops):
+        total = 0
+        for i in range(len(threads)):
+            total += results[i]
+        delta = total-lasttotal
+        speedkBps = (delta/sleepseconds)/(1024)
+        if verbose:
+            total_mb = total / (1024 * 1024)
+            delta_mb = delta / (1024 * 1024)
+            net_bits = application_bytes_to_networkbits(speedkBps) / 1024
+            mbps = "%.1f" % net_bits
+            print("Loop", loop, "Total MB", total_mb, "Delta_MB", delta_mb)
+            print("Speed kB/s:", speedkBps, "aka Mbps ", mbps)
+            lasttotal = total
```

Comment:
This is the source of the bug which leads to a spurious (high) result. This line should be outside the `if verbose:` block, semantically speaking I'd put it at the end of the loop, just before `time.sleep()`.